eurocodedesign/units/__init__.py

```diff
@@ -0,0 +1,255 @@
+        # check other order because tuple dict keys are ordered
+        if not new_type:
+            new_type = _allowed_multiplications.get(
+                (other_type, self_type), False)
+        if not new_type:
+            raise TypeError(f"Multiplication not allowed for"
+                            f" type {self_type} and {other_type}.")
+        return new_type(self._value * other._value)
+
+    def __rmul__(self, other):
+        return self * other
+
+    def __truediv__(self, other: Self | float | int):
+        if isinstance(other, (int, float)):
+            return type(self)(self._value
+                              / other
+                              / (self._prefix.value
+                                 ** self._power), self._prefix)
+        self_type = type(self)
+        other_type = type(other)
+
+        if self_type == other_type:
+            return float(self._value/other._value)
+
+        if self_type not in _allowed_multiplications.values():
+            raise TypeError('No divisions by another unit allowed'
+                            f' for f{self_type}')
+        keys = [k for k, v in _allowed_multiplications.items()
+                if v == self_type]
+        for type_pair in keys:
+            if other_type in type_pair:
+                l_ = list(type_pair)
+                l_.remove(other_type)
+                new_type = l_[0]
+                return new_type(self._value / other._value)
+        raise TypeError(f'Division of {self_type} by '
+                        f'{other_type} not allowed.')
+
+    def __lt__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return self._value < other._value
+
+    def __le__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return self._value <= other._value
+
+    def __eq__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return self._value == other._value
+
+    def __ne__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return not (self == other)
+
+    def __ge__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return self._value >= other._value
+
+    def __gt__(self, other: Self):
+        if (type(self) != type(other) or
+            self._physical_type != other._physical_type):
+            raise TypeError
+        return self._value > other._value
+
+    def to(self, prefix: Prefix):
+        """
+        Convert unit prefix to given prefix
+        @param prefix: Prefix,
+        @return: None
+        """
```

Comment:
We should be  consistent with our doc strings. this is what I have used : https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Can discuss.

Reply:
I did a little research and will you google style, too. We should add that to our documentation.